### PR TITLE
Build runnable alfacase built with SCORE data

### PR DESCRIPTION
**Pull request by gabrielantao**
Configure some default values for configured case in order to have a runnable case. Functions to parse result data and extract to csv were added.

NOTE: The current case configured from example (in the tests folder) is running, but it generates dubious results, different of the SCORE results for pressure and temperature (see the Jira task for the plots).

PWPA-1995

### CODE REVIEW

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 1]
```diff
@@ -312,3 +313,35 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
             "api_gravity": Scalar(fluid_data["api_gravity"], FRACTION_UNIT),
             "gas_gravity": Scalar(fluid_data["gas_gravity"], FRACTION_UNIT),
         }
+
+    def read_output_curves(self) -> Dict[str, Array]:
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py:317]: Include a comment in both methods, that they are only use to "test", compare results quem presented in json.

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 2]
```diff
@@ -312,3 +313,35 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
...
+        """Get the results from the input file."""
+        result = self.input_content["operation"]["thermal_simulation"]["result"]["0"]
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py]: Here you are expecting that it exist, I'd rather use get, and check if it exist.

File: src/alfasim_score/converter/alfacase/base_operation.py [hunk 3]
```diff
@@ -123,7 +208,65 @@ def build_nodes(self) -> List[NodeDescription]:
...
+                            FLUID_GAS: Array([0.0], VELOCITY_UNIT),
+                            FLUID_OIL: Array([0.0], VELOCITY_UNIT),
+                            FLUID_WATER: Array([0.0], VELOCITY_UNIT),
+                        },
+                    ),
+                ),
+                temperatures=InitialTemperaturesDescription(
+                    position_input_type=TableInputType.length,
+                    table_length=TemperaturesContainerDescription(
+                        positions=Array([well_length.GetValue()], LENGTH_UNIT),
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/base_operation.py]: Here you could use the formation temperature.
Keep, the bottom position/temperature as it was included, however include the top position (0 m) and the top temperature as defined in environment.

File: src/alfasim_score/converter/alfacase/base_operation.py [hunk 4]
```diff
@@ -123,7 +208,65 @@ def build_nodes(self) -> List[NodeDescription]:
 
     def build_well(self) -> WellDescription:
         """Create the description for the well."""
+        well_length = self.get_position_in_well(self.score_input.read_general_data()["final_md"])
+        operation_data = self.score_input.read_operation_data()
+        initial_bottom_pressure = operation_data["flow_initial_pressure"].GetValue(PRESSURE_UNIT)
+        # the factor multiplied for the top pressure is arbitrary, just to give a initial value
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/base_operation.py]: :+1:

File: src/alfasim_score/converter/alfacase/base_operation.py [hunk 5]
```diff
@@ -97,6 +172,16 @@ def build_nodes(self) -> List[NodeDescription]:
                     temperature=operation_data["flow_initial_temperature"],
                     pressure=operation_data["flow_initial_pressure"],
                     split_type=MassInflowSplitType.Pvt,
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/base_operation.py]: If you are using `split_type=MassInflowSplitType.Pvt,`, you don't need to define `volume_fraction`and `mass_fraction`.
`mass_fraction` is required when using: `MassInflowSplitType.ConstantMassFraction` and `volume_fraction`, when using: `MassInflowSplitType.ConstantVolumeFraction`

**Review by carlosnewmar — CHANGES_REQUESTED** [review]

**Review by lincrosenbach — APPROVED** [review]

File: src/alfasim_score/converter/alfacase/base_operation.py (hunk 3 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/base_operation.py]: Ok, sounds good, done.

File: src/alfasim_score/converter/alfacase/base_operation.py (hunk 5 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/base_operation.py]: Alright. Removed.

File: src/alfasim_score/converter/alfacase/base_operation.py (hunk 4 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/base_operation.py]: 😄

File: src/alfasim_score/converter/alfacase/score_input_reader.py (hunk 2 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/score_input_reader.py]: Indeed, I check now if I have result.

File: src/alfasim_score/converter/alfacase/score_input_reader.py (hunk 1 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/score_input_reader.py:317]: Ok, added the comment

File: src/alfasim_score/converter/alfacase/base_operation.py (hunk 3 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/base_operation.py]: With this change the case converges ok.

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 6]
```diff
@@ -312,3 +313,45 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
             "api_gravity": Scalar(fluid_data["api_gravity"], FRACTION_UNIT),
             "gas_gravity": Scalar(fluid_data["gas_gravity"], FRACTION_UNIT),
         }
+
+    def read_output_curves(self) -> Dict[str, Array]:
+        """
+        Get the results from the input file.
+        This parsed output curves are used to check cases results.
+        """
+        if "result" in self.input_content["operation"]["thermal_simulation"]:
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py:322]: A more "fancie" approach could be:
```
if results := self.input_content["operation"]["thermal_simulation"].get("result"):
    result = results["0"]
```

**Review by carlosnewmar — APPROVED** [review]

File: src/alfasim_score/converter/alfacase/score_input_reader.py (hunk 6 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/score_input_reader.py:322]: Honestly, I think this syntax is terrible hahahhaha.

**Pull request merged**